src/codegen/compile.ml

```diff
@@ -101,6 +101,15 @@ module Const = struct
     | Float64 of Numerics.Float.t
     | Blob of string
 
+  let lit_eq = function
+    | (Vanilla i, Vanilla j) -> i = j
+    | (BigInt i, BigInt j) -> Big_int.eq_big_int i j
+    | (Word32 i, Word32 j) -> i = j
+    | (Word64 i, Word64 j) -> i = j
+    | (Float64 i, Float64 j) -> i = j
```

Comment:
this seems fishy, maybe we shouldn't use `=` for this?

Reply:
Just as fishy as matching on float literals in the first place?

Reply:
Does `Numerics.Float.t` provide a `=` at all?